workflow/validate/validate.go

```diff
@@ -104,24 +105,59 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	ctx := newTemplateValidationCtx(wf, opts)
 	tmplCtx := templateresolution.NewContext(wftmplGetter, cwftmplGetter, wf, wf)
 
-	err := validateWorkflowFieldNames(wf.Spec.Templates)
+	var wftmpl wfv1.WorkflowSpecHolder
+	var entrypoint = wf.Spec.Entrypoint
+	var topLevelTmplRef *wfv1.TemplateRef
+	var err error
+
+	hasWorkflowTemplateRef := wf.Spec.WorkflowTemplateRef != nil
+
+	if hasWorkflowTemplateRef {
+		err := ValidateFieldsWithWFTRef(wf.Spec)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if hasWorkflowTemplateRef {
```

Comment:
Why are there two identical `if hasWorkflowTemplateRef {` one right after the other? Should you join them together?